prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -312,6 +312,18 @@ class ActionHistory(
         action.externalName = payloadName
     }
 
+    /**
+     * Adds information to the Action object about the organization and receiver channel affected by this action.
+     * Typically, this would be called when a report is batched for that receiver, sent to that receiver,
+     * downloaded by that receiver, or any other action taken by that receiver or on behalf of that receiver.
+     * @param organizationName  The name of the receiving organization to associate with this action.
+     * @param receiverName  The name of the receiver channel to associate with this action.
+     */
+    fun trackActionReceiverInfo(organizationName: String, receiverName: String) {
+        action.receivingOrg = organizationName
+        action.receivingOrgSvc = receiverName
+    }
+
```

Comment:
This is the meat of the change:  addition of columns to the Action table that let us track the receivingOrg.

This already exists for the sendingOrg side, so we are just adding it for the receivingOrg, so that our receiving and sending history APIs can have parallel code.

You could make a strong case that these various Org/Sender/Receiver string values should always have been in the Action table, and not the Report_file table, because putting these values in the action table lets us track info for error cases, when no report_file entry is created.